lib/ansible/modules/source_control/git_config.py

```diff
@@ -212,13 +230,19 @@ def main():
             k, v = value.split('=', 1)
             config_values[k] = v
         module.exit_json(changed=False, msg='', config_values=config_values)
-    elif not new_value:
+    elif not new_value and not unset:
         module.exit_json(changed=False, msg='', config_value=out.rstrip())
+    elif unset and not out:
+        module.exit_json(changed=False, msg='no setting to unset')
     else:
         old_value = out.rstrip()
         if old_value == new_value:
             module.exit_json(changed=False, msg="")
 
+    if unset:
+        args.insert(len(args) - 1, "--" + unset)
+        new_value = ''  # No new value when unset
```

Comment:
This will still result in a command like `git config --unset foo ''`.  According to the git config man page, that extra argument is a "value_regex" and its presence means only those values matching this regex will be unset.

Luckily the empty string is a regex that matches everything, so it all works out fine in the end.

Reply:
You're right. I didn't see it.
Taken into account.